From 3e6ad2c35232fe412dd7087751a8d9a5e42f8274 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Fri, 19 Aug 2022 09:11:12 +0300
Subject: [PATCH] [SW-86869] gaudi3: remove extra check to enter erom recovery

ROM recovery doesn't support secure firmware update and opens
all interfaces for debug. This patch removes OOB protocol and I3C
enable fuse checks for determining recovery flow and will be decided
only on recovery location efuse bit.

Change-Id: I957864ecf2ec562652e19b80c01aa507715606ad
---

diff --git a/arch/gaudi3/arc/recovery.c b/arch/gaudi3/arc/recovery.c
index 007cddb..572f735 100644
--- a/arch/gaudi3/arc/recovery.c
+++ b/arch/gaudi3/arc/recovery.c
@@ -237,16 +237,8 @@
 	 * efuse[EFUSE_SYSCFG_RECOVERY_CODE_LOCATION_MASK]:
 	 * 0: eROM
 	 * 1: ROM
-	 * efuse[EFUSE_SYSCFG_OOB_PROTOCOL]:
-	 * 0: NVMe (Legacy mode) -> ROM
-	 * 1: PLDM -> eROM
-	 * efuse[EFUSE_SYSCFG_I3C_EROM_ENABLE_MASK]:
-	 * 0: No I3C -> ROM
-	 * 1: I3C -> eROM
 	 */
-	if ((!(arcboot_rom_global.config_fuse & EFUSE_SYSCFG_RECOVERY_CODE_LOCATION_MASK)) ||
-		(arcboot_rom_global.config_fuse & EFUSE_SYSCFG_OOB_PROTOCOL) ||
-		(arcboot_rom_global.config_fuse & EFUSE_SYSCFG_I3C_EROM_ENABLE_MASK))
+	if (!(arcboot_rom_global.config_fuse & EFUSE_SYSCFG_RECOVERY_CODE_LOCATION_MASK))
 		execute_erom_recovery();
 	else
 		execute_rom_recovery();
